### ISSUE TITLE
RN 0.59.10  Unable to resolve module `AccessibilityInfo` .... ..... Module `AccessibilityInfo` does not exist in the Haste module map

### ISSUE THREAD

**Issue by git-single-lee** [facebook/react-native#28495]
BUNDLE  [dev] ./index.js ▓▓▓▓▓░░░░░░░░░░░ 36.0% (9/15)::1 - - [02/Apr/2020:15:50:07 +0000] "GET /index.bundle HTTP/1.1" 500 - "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36"
error: bundling failed: Error: **Unable to resolve module `AccessibilityInfo` from `/Users/coolqi/GitHubCode/rnText/node_modules/react-native/Libraries/react-native/react-native-implementation.js`: Module `AccessibilityInfo` does not exist in the Haste module map**

This might be related to https://github.com/facebook/react-native/issues/4968
To resolve try the following:
  1. Clear watchman watches: `watchman watch-del-all`.
  2. Delete the `node_modules` folder: `rm -rf node_modules && npm install`.
  3. Reset Metro Bundler cache: `rm -rf /tmp/metro-bundler-cache-*` or `npm start -- --reset-cache`.
  4. Remove haste cache: `rm -rf /tmp/haste-map-react-native-packager-*`.
    at ModuleResolver.resolveDependency (/Users/coolqi/GitHubCode/rnText/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:183:15)
    at ResolutionRequest.resolveDependency (/Users/coolqi/GitHubCode/rnText/node_modules/metro/src/node-haste/DependencyGraph/ResolutionRequest.js:52:18)
    at DependencyGraph.resolveDependency (/Users/coolqi/GitHubCode/rnText/node_modules/metro/src/node-haste/DependencyGraph.js:283:16)
    at Object.resolve (/Users/coolqi/GitHubCode/rnText/node_modules/metro/src/lib/transformHelpers.js:261:42)
    at /Users/coolqi/GitHubCode/rnText/node_modules/metro/src/DeltaBundler/traverseDependencies.js:399:31
    at Array.map (<anonymous>)
    at resolveDependencies (/Users/coolqi/GitHubCode/rnText/node_modules/metro/src/DeltaBundler/traverseDependencies.js:396:18)
    at /Users/coolqi/GitHubCode/rnText/node_modules/metro/src/DeltaBundler/traverseDependencies.js:269:33
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/coolqi/GitHubCode/rnText/node_modules/metro/src/DeltaBundler/traverseDependencies.js:87:24)
 BUNDLE  [dev] ./index.js ▓▓▓▓▓░░░░░░░░░░░ 36.0% (9/15), failed.

**Comment by git-single-lee** [comment]
$ /Users/coolqi/GitHubCode/rnText/node_modules/.bin/react-native info
info 
  React Native Environment Info:
    System:
      OS: macOS 10.15.3
      CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
      Memory: 719.35 MB / 32.00 GB
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.4.0 - /var/folders/q_/z79stvl533n3v9jwm2zn9rmr0000gn/T/yarn--1585842932386-0.5969058946125583/node
      Yarn: 1.22.4 - /var/folders/q_/z79stvl533n3v9jwm2zn9rmr0000gn/T/yarn--1585842932386-0.5969058946125583/yarn
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      Android SDK:
        API Levels: 23, 25, 26, 27, 28, 29
        Build Tools: 25.0.0, 26.0.2, 26.0.3, 27.0.3, 28.0.2, 28.0.3, 29.0.0, 29.0.2
        System Images: android-27 | Google Play Intel x86 Atom
    IDEs:
      Xcode: /undefined - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3 
      react-native: 0.59.10 => 0.59.10 

✨  Done in 3.85s.

**Comment by git-single-lee** [comment]
真他妈的，垃圾，这破框架
谁知道这个该怎么解决，我看到其他人也提了这个问题，但是被莫名其妙的关闭了。。

Shit, trash, this broken frame
Who knows how to solve this, I saw other people also raised this issue, but it was inexplicably closed. .

**Comment by rickhanlonii** [comment]
@git-single-lee, thanks for filing the issue.

I've removed the first line of your post to protect people who may actually struggle with those thoughts. If that is a serious concern for you, please reach out to me directly or [other resources for support](https://suicidepreventionlifeline.org/).

**Comment by git-single-lee** [comment]
i deal with this...

**Comment by rickhanlonii** [comment]
@cpojer @motiz88 @rubennorte this issue is super common, is there anything we can do to improve this error or catch common cases and recommend solutions? It seems like the haste map is just frequently wrong.

**Comment by safaiyeh** [comment]
Is this still an issue in 0.63? @git-single-lee @rickhanlonii

**Comment by OKNoah** [comment]
I have this issue with `0.63.4`, and I've been reviewing reports of this error all the way back to `0.44.1`. Unfortunately, the consensus seems to always be that each version is simply broken and the solution is to upgrade. The issues are usually closed as resolve, stale or spam before any other solution or cause is explored. Unless the maintainers are _repeatedly_ breaking the project in the same way over the course of years, I think it's safe to say that it's not really a problem with countless versions, and that upgrading is simply having the side effect of fixing some other local problem.

Any thoughts or trouble shooting ideas for this besides upgrading?

**Comment by ohrl** [comment]
Add 'platform=ios' or 'platform=android' in url. Because 'index.js' import 'AccessibilityInfo', but in './Libraries/Components/AccessibilityInfo/' only have files named 'AccessibilityInfo.ios.js'/'AccessibilityInfo.android.js'

**Comment by SliverYuki** [comment]
Is there any update? I am encountering the same problem.
